File: mobile/lib/data/models/pharmacy_model.dart
class PharmacyModel {
  final String id;
  final String pharmacyName;
  final String businessRegNumber;
  final String permitNumber;
  final String province;
  final String district;
  final String city;
  final String address;
  final String postalCode;
  final String ownerName;
  final String ownerNIC;
  final String email;
  final String status;
  final String? rejectionReason;

  const PharmacyModel({
    required this.id,
    required this.pharmacyName,
    required this.businessRegNumber,
    required this.permitNumber,
    required this.province,
    required this.district,
    required this.city,
    required this.address,
    required this.postalCode,
    required this.ownerName,
    required this.ownerNIC,
    required this.email,
    required this.status,
    this.rejectionReason,
  });

  factory PharmacyModel.fromJson(Map<String, dynamic> json) {
    return PharmacyModel(
      id: json['id'] as String? ?? json['_id'] as String? ?? '',
      pharmacyName: json['pharmacyName'] as String? ?? '',
      businessRegNumber: json['businessRegNumber'] as String? ?? '',
      permitNumber: json['permitNumber'] as String? ?? '',
      province: json['province'] as String? ?? '',
      district: json['district'] as String? ?? '',
      city: json['city'] as String? ?? '',
      address: json['address'] as String? ?? '',
      postalCode: json['postalCode'] as String? ?? '',
      ownerName: json['ownerName'] as String? ?? '',
      ownerNIC: json['ownerNIC'] as String? ?? '',
      email: json['email'] as String? ?? '',
      status: json['status'] as String? ?? 'pending',
      rejectionReason: json['rejectionReason'] as String?,
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'pharmacyName': pharmacyName,
        'businessRegNumber': businessRegNumber,
        'permitNumber': permitNumber,
        'province': province,
        'district': district,
        'city': city,
        'address': address,
        'postalCode': postalCode,
        'ownerName': ownerName,
        'ownerNIC': ownerNIC,
        'email': email,
        'status': status,
        if (rejectionReason != null) 'rejectionReason': rejectionReason,
      };
}

File: backend/src/models/pharmacy.model.js
const mongoose = require('mongoose');

const pharmacySchema = new mongoose.Schema({
  pharmacyName: {
    type: String,
    required: true,
    trim: true,
  },

  businessRegNumber: {
    type: String,
    required: true,
    unique: true,
    trim: true,
  },

  permitNumber: {
    type: String,
    required: true,
    trim: true,
  },

  // Location
  province: {
    type: String,
    required: true,
  },

  district: {
    type: String,
    required: true,
  },

  city: {
    type: String,
    required: true,
    trim: true,
  },

  address: {
    type: String,
    required: true,
    trim: true,
  },

  postalCode: {
    type: String,
    required: true,
    trim: true,
  },

  // Owner
  ownerName: {
    type: String,
    required: true,
    trim: true,
  },

  ownerNIC: {
    type: String,
    required: true,
    unique: true,
    trim: true,
  },

  email: {
    type: String,
    required: true,
    unique: true,
    lowercase: true,
    trim: true,
  },

  password: {
    type: String,
    required: true,
  },

  status: {
    type: String,
    enum: ['pending', 'approved', 'rejected'],
    default: 'pending',
  },

  bankDetails: {
    bankName:          { type: String, default: null },
    branchName:        { type: String, default: null },
    accountNumber:     { type: String, default: null },
    accountHolderName: { type: String, default: null },
    accountType:       { type: String, enum: ['Savings', 'Current', null], default: null },
  },

  rejectionReason: {
    type: String,
    default: null,
  },

  createdAt: {
    type: Date,
    default: Date.now,
  },
});

module.exports = mongoose.model('Pharmacy', pharmacySchema);
